### ISSUE TITLE
Debugging doesn't work in 1.19.0-dev.7.3 SDK

### ISSUE THREAD

**Issue by DanTup** [Dart-Code/Dart-Code#129]
Using 1.19 RC SDK.

Code:

```dart
main() {
  print("test");
}
```

Press `F5` to run under debugger; program never runs. Logged output is:

```javascript
==> {"id":"0","method":"streamListen","params":{"streamId":"Isolate"}}
==> {"id":"1","method":"streamListen","params":{"streamId":"Debug"}}
<== {"jsonrpc":"2.0","id":"0","result":{"type":"Success"}}
<== {"jsonrpc":"2.0","id":"1","result":{"type":"Success"}}
```

Whereas in 1.18.1 we see:

```
==> {"id":"0","method":"streamListen","params":{"streamId":"Isolate"}}
==> {"id":"1","method":"streamListen","params":{"streamId":"Debug"}}
<== {"jsonrpc":"2.0","id":"0","result":{"type":"Success"}}
<== {"jsonrpc":"2.0","id":"1","result":{"type":"Success"}}
<== {"jsonrpc":"2.0","method":"streamNotify","params":{"streamId":"Isolate","event":{"type":"Event","kind":"IsolateRunnable","isolate":{"type":"@Isolate","fixedId":true,"id":"isolates\/243414459","name":"test.dart$main","number":"243414459"},"timestamp":1472205353653}}}
==> {"id":"2","method":"setExceptionPauseMode","params":{"isolateId":"isolates/243414459","mode":"Unhandled"}}
<== {"jsonrpc":"2.0","method":"streamNotify","params":{"streamId":"Debug","event":{"type":"Event","kind":"PauseStart","isolate":{"type":"@Isolate","fixedId":true,"id":"isolates\/243414459","name":"test.dart$main","number":"243414459"},"timestamp":1472205353655}}}
==> {"id":"3","method":"resume","params":{"isolateId":"isolates/243414459"}}
<== {"jsonrpc":"2.0","method":"streamNotify","params":{"streamId":"Debug","event":{"type":"Event","kind":"_DebuggerSettingsUpdate","isolate":{"type":"@Isolate","fixedId":true,"id":"isolates\/243414459","name":"test.dart$main","number":"243414459"},"timestamp":1472205353657,"_debuggerSettings":{"_exceptions":"unhandled"}}}}
```

The 1.19 version never gets the IsolateRunnable event that came through in 1.18?

**Comment by DanTup** [comment]
@devoncarew This is actually really weird.

In 1.18.1 SDK, the breakpoint doesn't get hit:

```shell
dart test.dart
==> {"id":"0","method":"streamListen","params":{"streamId":"Isolate"}}
==> {"id":"1","method":"streamListen","params":{"streamId":"Debug"}}
<== {"jsonrpc":"2.0","id":"0","result":{"type":"Success"}}
<== {"jsonrpc":"2.0","id":"1","result":{"type":"Success"}}
<== {"jsonrpc":"2.0","method":"streamNotify","params":{"streamId":"Isolate","event":{"type":"Event","kind":"IsolateRunnable","isolate":{"type":"@Isolate","fixedId":true,"id":"isolates\/33279243","name":"test.dart$main","number":"33279243"},"timestamp":1472155731044}}}
==> {"id":"2","method":"setExceptionPauseMode","params":{"isolateId":"isolates/33279243","mode":"Unhandled"}}
==> {"id":"3","method":"addBreakpointWithScriptUri","params":{"isolateId":"isolates/33279243","scriptUri":"file:///m:/Apps/Dart/lib/core/print.dart","line":9}}
<== {"jsonrpc":"2.0","method":"streamNotify","params":{"streamId":"Debug","event":{"type":"Event","kind":"PauseStart","isolate":{"type":"@Isolate","fixedId":true,"id":"isolates\/33279243","name":"test.dart$main","number":"33279243"},"timestamp":1472155731045}}}
<== {"jsonrpc":"2.0","method":"streamNotify","params":{"streamId":"Debug","event":{"type":"Event","kind":"_DebuggerSettingsUpdate","isolate":{"type":"@Isolate","fixedId":true,"id":"isolates\/33279243","name":"test.dart$main","number":"33279243"},"timestamp":1472155731047,"_debuggerSettings":{"_exceptions":"unhandled"}}}}
<== {"jsonrpc":"2.0", "result":{"type":"Success"},"id":"2"}
<== {"jsonrpc":"2.0","method":"streamNotify","params":{"streamId":"Debug","event":{"type":"Event","kind":"BreakpointAdded","isolate":{"type":"@Isolate","fixedId":true,"id":"isolates\/33279243","name":"test.dart$main","number":"33279243"},"timestamp":1472155731049,"breakpoint":{"type":"Breakpoint","fixedId":true,"id":"breakpoints\/1","breakpointNumber":1,"resolved":false,"location":{"type":"UnresolvedSourceLocation","scriptUri":"file:\/\/\/m:\/Apps\/Dart\/lib\/core\/print.dart","line":9}}}}}
<== {"jsonrpc":"2.0", "result":{"type":"Breakpoint","fixedId":true,"id":"breakpoints\/1","breakpointNumber":1,"resolved":false,"location":{"type":"UnresolvedSourceLocation","scriptUri":"file:\/\/\/m:\/Apps\/Dart\/lib\/core\/print.dart","line":9}},"id":"3"}
==> {"id":"4","method":"resume","params":{"isolateId":"isolates/33279243"}}
<== {"jsonrpc":"2.0","method":"streamNotify","params":{"streamId":"Debug","event":{"type":"Event","kind":"Resume","isolate":{"type":"@Isolate","fixedId":true,"id":"isolates\/33279243","name":"test.dart$main","number":"33279243"},"timestamp":1472155731050}}}
<== {"jsonrpc":"2.0", "result":{"type":"Success"},"id":"4"}
test
<== {"jsonrpc":"2.0","method":"streamNotify","params":{"streamId":"Isolate","event":{"type":"Event","kind":"IsolateExit","isolate":{"type":"@Isolate","fixedId":true,"id":"isolates\/33279243","name":"test.dart$main","number":"33279243"},"timestamp":1472155731051}}}
```

And in 1.19 it seems to pause early on, and almost no data goes over to the wire to the VM:

```shell
dart test.dart
==> {"id":"0","method":"streamListen","params":{"streamId":"Isolate"}}
==> {"id":"1","method":"streamListen","params":{"streamId":"Debug"}}
<== {"jsonrpc":"2.0","id":"0","result":{"type":"Success"}}
<== {"jsonrpc":"2.0","id":"1","result":{"type":"Success"}}
```

Do you know if there are expected changes to the protocol? The 1.19 version never gets the `IsolateRunnable` event that came through in 1.18?

**Comment by devoncarew** [comment]
I don't know that the protocol has changed much recently. It's odd that the vm versions behave differently -

**Comment by rmacnak-google** [comment]
I don't see a IsolateRunnable in 1.18.1 either. I think this is a usage error, with the intended usage being

1) Subscribe to the isolate stream to learn about new isolates.
2) Call getVM learn about the isolates that were already running.

@turnidge

**Comment by DanTup** [comment]
Maybe this is a timing issue then? Maybe in 1.18.1 our subscription is being set up sooner? It's 100% reproducible for me though, those logs I posted are what I see for each version every time.

**Comment by DanTup** [comment]
Confirmed fixed by #132 :)
